py3status/modules/prometheus.py

```diff
@@ -94,7 +104,12 @@ def prometheus(self):
         return ret
 
     def _query(self, query):
-        r = self.py3.request(self.server + "/api/v1/query", params={"query": query})
+        try:
+            r = self.py3.request(self.server + "/api/v1/query", params={"query": query})
+        except Exception as e:
+            self.py3.log(f"prometheus: Request failed {e}")
+            return []
```

Comment:
```python
prometheus {
    on_error = "hide"
}
``` 
... might be what you want in your config... It retries every 10 seconds, I think.